akka-remote/src/main/scala/akka/remote/artery/LruBoundedCache.scala

```diff
@@ -0,0 +1,211 @@
+          values(position)
+        } else {
+          // This is not our slot yet
+          findOrCalculate((position + 1) & Mask, probeDistance + 1)
+        }
+      }
+    }
+
+    findOrCalculate(position = h & Mask, probeDistance = 0)
+  }
+
+  @tailrec private def removeAt(position: Int): Unit = {
+    val next = (position + 1) & Mask
+    if ((values(next) eq null) || probeDistanceOf(next) == 0) {
+      // Next is not movable, just empty this slot
+      values(position) = null.asInstanceOf[V]
+    } else {
+      // Shift the next slot here
+      keys(position) = keys(next)
+      values(position) = values(next)
+      hashes(position) = hashes(next)
+      epochs(position) = epochs(next)
+      // remove the shifted slot
+      removeAt(next)
+    }
+  }
+
+  // Wraparound distance of the element that is in this slot. (X + capacity) & Mask ensures that there are no
+  // negative numbers on wraparound
+  private def probeDistanceOf(slot: Int): Int = probeDistanceOf(idealSlot = hashes(slot) & Mask, actualSlot = slot)
+
+  // Protected for exposing it to unit tests
+  protected def probeDistanceOf(idealSlot: Int, actualSlot: Int) = ((actualSlot - idealSlot) + capacity) & Mask
+
+  @tailrec private def move(position: Int, k: K, h: Int, value: V, elemEpoch: Int, probeDistance: Int): Unit = {
+    if (values(position) eq null) {
+      // Found an empty place, done.
+      keys(position) = k
+      values(position) = value
+      hashes(position) = h
+      epochs(position) = elemEpoch // Do NOT update the epoch of the elem. It was not touched, just moved
+    } else {
+      val otherEpoch = epochs(position)
+      // Check if the current entry is too old
+      if (epoch - otherEpoch >= evictAgeThreshold) {
+        // Remove old entry to make space
+        removeAt(position)
+        // Try to insert our element in hand to its ideal slot
+        move(h & Mask, k, h, value, elemEpoch, 0)
+      } else {
+        val otherProbeDistance = probeDistanceOf(position)
+        val otherEpoch = epochs(position)
+
+        // Check whose probe distance is larger. The one with the larger one wins the slot.
+        if (probeDistance > otherProbeDistance) {
+          // Due to the Robin-Hood property, we now take away this slot from the "richer" and take it for ourselves
+          val otherKey = keys(position)
+          val otherValue = values(position)
+          val otherHash = hashes(position)
+
+          keys(position) = k
+          values(position) = value
+          hashes(position) = h
+          epochs(position) = elemEpoch
+
+          // Move out the old one
+          move((position + 1) & Mask, otherKey, otherHash, otherValue, otherEpoch, otherProbeDistance + 1)
+        } else {
+          // We are the "richer" so we need to find another slot
+          move((position + 1) & Mask, k, h, value, elemEpoch, probeDistance + 1)
+        }
+
+      }
+    }
+
+  }
+
+  protected def calculate(k: K): V
```

Comment:
very minor: would be nicer as "compute" to play along with the getOrCompute wording